client/src/game/reducers/integration.test.ts

```diff
@@ -130,6 +158,34 @@ describe('integration', () => {
         expect(finalState.stats.potentialCluesLost).toBe(18);
       });
     });
+
+    describe.each([...Array(rainbowOnesAndPinkGame.actions.length).keys()])('on turn %i', (turn) => {
+      test.each([...Array(45).keys()])(
```

Comment:
this is not an integration test - would be better to get it as a unit test with a single turn, with a few carefully chosen cards.
an integration test for this is what we're doing now, which is to compare an expected final state with the actual result we got.

Reply:
My intention was for it to be more of a sanity check - making sure the logic doesn't mess up so badly that it somehow removes the correct suit/rank.  I was also thinking about running through the game for each player (so removing knowledge of your own hand from the underlying logic) with the same check.  It's not really making a positive check (saying "it removed this correct suit/rank"), so maybe that's not valuable :/.  I can remove it.

For unit tests: I added some tests under `cardsReducer.test`.  I sort of treated it as though `cardDeductionReducer` was a part of `cardsReducer` when running tests but I can see an argument that it's not really a unit test at that point.  Would it be best to mock the `cardDeductionReducer` in `cardsReducer.test` and just move those tests over to a `cardDeductionReducer.test` (asserting a single turn)?

Reply:
I believe these "on turn %i" tests have been removed.